### PR TITLE
Allow the Identity Provider to be the primary auth method in Harbor.

### PR DESCRIPTION
**Pull request by tpoxa**
New **optional** feature:  Primary Auth autoselect.

When primary auth is enabled, The IDP becomes the default or primary method for users to login into Harbor. 
When the users enter the base URL of the Harbor instance, the login screen where the user can choose between login via the identity provider or via local DB is not shown. The user will be automatically redirected to the identity provider (when primary auth is enabled in config). Login via DB is still possible. To do so, the Harbor admin needs to visiting the login url '/account/sign-in' explicitly.

The functionality can be enabled/disabled in the Harbor config and hence is optional.

### Demo that shows how primary auth is working

https://user-images.githubusercontent.com/1492007/203007280-01db3185-3202-4748-a59c-5807a961e3ca.mp4

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. **release-note/new-feature**
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by wy65701436** [comment]
@tpoxa can you please help to give more details on this PR?

**Comment by tpoxa** [comment]
@wy65701436 well, oidc users always have to make an extra click time during sign-in; when this option is on, UI should redirect to the oidc provider directly. That was the plan.

File: src/portal/src/i18n/lang/de-de-lang.json [hunk 1]
```diff
@@ -881,6 +881,7 @@
         "VERIFY_REMOTE_CERT": "Zertifikat der Gegenstelle prüfen",
         "TOKEN_EXPIRATION": "Token-Ablauf (Minuten)",
         "AUTH_MODE": "Auth Mode",
+        "PRIMARY_AUTH_MODE": "Dieser Anmeldemodus wird zur Standardmethode für die Benutzeranmeldung. Der Anmeldebildschirm, in dem der Benutzer auswählt, ob er sich über den Identitätsanbieter oder über die lokale DB anmelden möchte, leitet den Benutzer automatisch an diesen Identitätsanbieter weiter. Die Anmeldung über die DB ist möglich, wenn die URL '/account/sign-in' explizit aufgerufen wird.",
```
Comment (Vad1mo) [inline src/portal/src/i18n/lang/de-de-lang.json:884]: This is an error. Here shouldn't be the text for the tooltip. It should be the label name: **Primärer Anmeldemodus**

Comment (Vad1mo) [reply src/portal/src/i18n/lang/de-de-lang.json:884]: L905 after auth_mode should contain the tooltip.


`        "PRIMARY_AUTH_MODE": "Dieser Anmeldemodus wird zur Standardmethode für die Benutzeranmeldung. Der Anmeldebildschirm, in dem der Benutzer auswählt, ob er sich über den Identitätsanbieter oder über die lokale DB anmelden möchte, leitet den Benutzer automatisch an diesen Identitätsanbieter weiter. Die Anmeldung über die DB ist möglich, wenn die URL '/account/sign-in' explizit aufgerufen wird."`

**Review by Vad1mo — CHANGES_REQUESTED** [review]
For the language files, where you can't provide a translation, please copy&paste the English text.

- es-es-lang.json
- fr-fr-lang.json
- pt-br-lang.json
- tr-tr-lang.json
- zh-cn-lang.json
- zh-tw-lang.json

**Review by Vad1mo — APPROVED** [review]
Looks good

**Comment by OrlinVasilev** [comment]
That's great!!! @Vad1mo @tpoxa

**Review by Vad1mo — APPROVED** [review]

**Comment by stonezdj** [comment]
LGTM

**Comment by OrlinVasilev** [comment]
@stonezdj can you approve again

**Review by stonezdj — APPROVED** [review]

**Comment by AllForNothing** [comment]
@tpoxa Please `cd src/portal`, then run `npm run lint_fix` to fix any lint-errors, then you can pass the UI-UT check.

**Comment by AllForNothing** [comment]
@Vad1mo I had a try on my own env.  This looks fine to me

**Review by AllForNothing — APPROVED** [review]
LGTM

**Review by Vad1mo — APPROVED** [review]

**Comment by till** [comment]
@yanji09 does this fix api calls for idp users as well?

**Comment by Rekhaprakash1** [comment]
@tpoxa @Vad1mo Is this feature available for API as well ?

**Pull request merged**